sipn_reanalysis_ingest/luigitasks/convert.py

```diff
@@ -39,139 +33,155 @@
 from sipn_reanalysis_ingest.util.misc import range_lookup
 
 
-class Grib2ToNcDaily(luigi.Task):
-    """Converts GRIB2 6-hourly input data to daily NetCDF.
+class Grib2ToDailyNc(luigi.Task):
+    """Shared logic and configuration for daily NetCDF processing."""
+
+    date = luigi.DateParameter()
+
+    def output(self):
+        fn = DATA_DAILY_FILENAME_TEMPLATE.format(date=self.date)
+        return luigi.LocalTarget(DATA_FINISHED_DIR / fn)
+
+    def convert(
+        self,
+        *,
+        analysis_inputs: list[Path],
+        forecast_inputs: list[Path],
+    ) -> None:
+        logger.info(f'Producing daily NetCDF for date {self.date}...')
+        logger.debug(f'>> Analysis inputs: {analysis_inputs}')
+        logger.debug(f'>> Forecast inputs: {forecast_inputs}')
+        with self.output().temporary_path() as tempf:
+            convert_6hourly_grib2s_to_nc(
+                analysis_inputs=analysis_inputs,
+                forecast_inputs=forecast_inputs,
+                output_path=Path(tempf),
+            )
+
+            for ifile in [*analysis_inputs, *forecast_inputs]:
+                ifile.unlink()
+
+
+class Grib2InDailyTarToDailyNc(Grib2ToDailyNc):
+    """Converts GRIB2 6-hourly input data in 1-daily tars to a daily NetCDF.
 
     The source data will be filtered for only the variables of interest, subset and
     reprojected for our area of interest.
     """
 
-    date = luigi.DateParameter()
+    def requires(self):
+        req = [
+            UntarCfsr1DayFile(date=self.date - dt.timedelta(days=1)),
+            UntarCfsr1DayFile(date=self.date),
+        ]
+        return req
 
-    @property
-    def tars_required(self) -> TarsRequiredForDailyData:
-        if self.date == CFSR_DAILY_TAR_ON_OR_AFTER:
-            return TarsRequiredForDailyData.BOTH
-        elif self.date < CFSR_DAILY_TAR_ON_OR_AFTER:
-            return TarsRequiredForDailyData.FIVE_DAILY
-        elif self.date > CFSR_DAILY_TAR_ON_OR_AFTER:
-            return TarsRequiredForDailyData.DAILY
```

Comment:
Moved this logic up one level in the dependency tree to the `luigitasks/main.py` module in the function `make_daily_processing_requirement`